services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/SwitchOperationsService.java

```diff
@@ -151,6 +151,8 @@ public boolean deleteSwitch(SwitchId switchId, boolean force) throws SwitchNotFo
             // forceDelete() removes switch along with all relationships.
             switchRepository.forceDelete(sw.getSwitchId());
         } else {
+            switchPropertiesRepository.findBySwitchId(sw.getSwitchId())
+                    .ifPresent(sp -> switchPropertiesRepository.delete(sp));
             // delete() is used to be sure that we wouldn't delete switch if it has even one relationship.
```

Comment:
Maybe we need to delete properties in `force` case too?

Reply:
Fixed